lib/Sema/TypeCheckAttr.cpp

```diff
@@ -2290,63 +2290,29 @@ void AttributeChecker::visitDifferentiableAttr(DifferentiableAttr *attr) {
     attr->setPrimalFunction(primal);
   }
 
-  // Compute the return type of the adjoint function.
-  auto wrtParams = attr->getParameters();
-  SmallVector<TupleTypeElt, 8> retElts;
+  // Validate the 'wrt:' parameters.
 
-  // If `type` is not allowed as a wrt type, diagnoses and returns true.
-  auto checkAndDiagnoseWrtType = [&](SourceLoc loc, Type type) -> bool {
-    if (type->isAnyClassReferenceType() || type->isExistentialType()) {
-      TC.diagnose(
-          loc,
-          diag::differentiable_attr_cannot_diff_wrt_objects_or_existentials,
-          type);
-      return true;
-    }
-    return false;
-  };
-
-  // If the self type of `original` is allowed as a wrt param, appends the
-  // corresponding return type to `retElts` and returns false. Otherwise,
-  // returns true and diagnoses.
-  auto addWrtSelfRetTyOrDiagnose = [&](SourceLoc loc) -> bool {
-    auto parent = original->getParent();
-    if (checkAndDiagnoseWrtType(loc, parent->getSelfTypeInContext()))
-      return true;
-    retElts.push_back(parent->getSelfInterfaceType());
-    return false;
-  };
+  // These are the wrt param indices specified by the user, which have not yet
+  // been checked.
+  auto uncheckedWrtParamIndices = attr->getParameters();
```

Comment:
They are not strictly indices, right? They are symbolic parameters because the type is a tagged union. I called it just "params".